playerbot/strategy/hunter/HunterActions.cpp

```diff
@@ -41,3 +41,23 @@ bool CastAutoShotAction::Execute(Event& event)
 
     return CastSpellAction::Execute(event);
 }
+
+bool CastSteadyShotAction::Execute(Event& event)
+{
+    bool castResult = CastSpellAction::Execute(event);
+
+    const Item* equippedWeapon = bot->GetItemByPos(INVENTORY_SLOT_BAG_0, EQUIPMENT_SLOT_RANGED);
+
+    if (equippedWeapon)
+    {
+        const ItemPrototype* itemPrototype = equippedWeapon->GetProto();
+        weaponDelay = itemPrototype->Delay + sPlayerbotAIConfig.globalCoolDown;
+
+        if (castResult == true)
+        {
+            SetDuration(weaponDelay);
+        }
+    }
+
+    return castResult;
+}

```

Comment:
all that is not needed if cast result is false

Reply:
Fixed